### PR TITLE
Auth 2867/remove v1 tokens gateway apis

### PR DESCRIPTION
**Pull request by blakestier**
### :nut_and_bolt: Description: What code changed, and why?

Removed v1 tokens paths/clients for force-upgrade to iam v2.

### CODE REVIEW

File: .studio/automate-gateway [hunk 1]
```diff
@@ -554,21 +554,19 @@ function get_api_token_with_policy() {
...
   else
     # create standard api token
-    chef-automate iam token create "$token_description" >"$token_file" || return 1
+    chef-automate iam token create "$token_name" >"$token_file" || return 1
     # retrieve the tokens' ID
     local token_id
     token_id="$(curl -fsS --insecure -H "api-token: $(get_admin_token)" \
-      "${GATEWAY_URL}/auth/tokens" |\
-        jq -er --arg token "$token_description" '[.tokens[] | select(.description == $token) | .id] | first' )"
+      "${GATEWAY_URL}/iam/v2/tokens" |\
```
Comment (bcmdarroch) [inline .studio/automate-gateway]: do we want the name here or the id? finding the jq a little hard to read 🤔

File: components/automate-cli/pkg/client/apiclient/apiclient.go [hunk 2]
```diff
@@ -25,7 +24,6 @@ type client struct {
 	authzClient        authz.AuthorizationClient
 	teamsClient        teams.TeamsClient
 	teamsV2Client      v2.TeamsClient
-	tokensClient       tokens.TokensMgmtClient
 	tokensV2Client     v2.TokensClient
```
Comment (bcmdarroch) [inline components/automate-cli/pkg/client/apiclient/apiclient.go]: let's drop `v2` where we can (including the `v2` import name up on L17)

```suggestion
	tokensClient    iam.TokensClient
```

File: components/automate-cli/pkg/client/mock/mock.go [hunk 3]
```diff
@@ -38,8 +36,7 @@ type ServerMocks struct {
 	PoliciesMock *v2.PoliciesServerMock
 	TeamsMock    *teams.TeamsServerMock
 	TeamsV2Mock  *v2.TeamsServerMock
-	TokensMock   *tokens.TokensMgmtServerMock
-	TokensV2Mock *v2.TokensServerMock
+	TokensMock   *v2.TokensServerMock
```
Comment (bcmdarroch) [inline components/automate-cli/pkg/client/mock/mock.go]:
```suggestion
	TokensMock   *iam.TokensServerMock
```

File: components/automate-cli/pkg/client/types.go [hunk 4]
```diff
@@ -18,7 +17,6 @@ type APIClient interface {
 	AuthzClient() authz.AuthorizationClient
 	TeamsClient() teams.TeamsClient
 	TeamsV2Client() v2.TeamsClient
-	TokensClient() tokens.TokensMgmtClient
 	TokensV2Client() v2.TokensClient
```
Comment (bcmdarroch) [inline components/automate-cli/pkg/client/types.go]:
```suggestion
	TokensClient() iam.TokensClient
```

File: components/automate-gateway/gateway/services.go [hunk 5]
```diff
@@ -356,7 +353,7 @@ func unversionedRESTMux(grpcURI string, dopts []grpc.DialOption) (http.Handler,
 		"gateway":              pb_gateway.RegisterGatewayHandlerFromEndpoint,
 		"legacy":               pb_legacy.RegisterLegacyDataCollectorHandlerFromEndpoint,
 		"license":              pb_license.RegisterLicenseHandlerFromEndpoint,
-		"auth tokens":          pb_tokens.RegisterTokensMgmtHandlerFromEndpoint,
+		"auth tokens":          pb_iam_v2.RegisterTokensHandlerFromEndpoint,
```
Comment (bcmdarroch) [inline components/automate-gateway/gateway/services.go]:
```suggestion
		"iam tokens":          pb_iam.RegisterTokensHandlerFromEndpoint,
```

File: components/automate-gateway/handler/tokens.go [hunk 6]
```diff
@@ -1,93 +0,0 @@
-package handler
-
-import (
```
Comment (bcmdarroch) [inline components/automate-gateway/handler/tokens.go]: where does the v2 tokens handler live?

File: components/automate-cli/pkg/client/apiclient/apiclient.go (hunk 2 above)
Comment (blakestier) [reply components/automate-cli/pkg/client/apiclient/apiclient.go]: Oh yeah I was gonna do that on a final pass. Thanks!

File: components/automate-gateway/handler/tokens.go (hunk 6 above)
Comment (blakestier) [reply components/automate-gateway/handler/tokens.go]: 💁 https://github.com/chef/automate/blob/master/components/automate-gateway/handler/iam/v2/tokens/tokens.go

File: .studio/automate-gateway (hunk 1 above)
Comment (blakestier) [reply .studio/automate-gateway]: I think we still want the id (to be added to the policy below)

File: components/automate-chef-io/content/docs/api-tokens.md [hunk 7]
```diff
@@ -36,13 +36,13 @@ Admin tokens are tokens that are automatically added to the Administrator policy
 Admin tokens can only be created using the `chef-automate` command line.
 
 ```
-chef-automate iam token create <your-token-name> --admin
+chef-automate iam token create <your-token-id> --admin
```
Comment (blakestier) [inline components/automate-chef-io/content/docs/api-tokens.md]: This command generates an id and name that match the value passed in. A conflict results in an error message about a conflicting id, not a conflicting name (we don't require uniqueness on name).

**Pull request merged**